Design/PDcalibration.md

```diff
@@ -0,0 +1,145 @@
+depending on whether operating on combined (in d-spacing) or
+individual pixels (in time-of-flight).
+
+Currently, much of the underlying peak finding/fitting is done in
+`FindPeaks`. It may be better to move to a customized algorithm that
+can make assumptions for powder diffraction. `FindPeaks` is the most
+time-consuming, error-prone, and confusing step in the calibration
+process.
+
+Pixel-by-pixel
+--------------
+
+This calibration will work by converting information from d-spacing to
+time-of-flight and determining the constants by calculating the
+conversion from d-spacing (known positions) to time-of-flight
+(observed positions).
+
+1. Calculate fit windows and parameters. This would take a list of
+   peaks in d-spacing, a calibration dataset, and a workspace
+   index. The result is information about (in time-of-flight) the
+   starting values for peak positions and fit windows. Ideally, this
+   would also provide information for starting peak widths and
+   heights. That is currently determined in `FindPeaks`.
+
+1. Find the actual peak positions. The output (per spectrum) would be
+   the peak positions, heights, and widths (in time-of-flight) as well
+   as the reduced chi-squared for the peak fit.
+
+1. Filter out spurious/unwanted observed peaks. The criteria in
+   [GetDetOffsetsMultiPeaks](http://docs.mantidproject.org/v3.5.1/algorithms/GetDetOffsetsMultiPeaks-v1.html#criteria-on-peaks)
+   should be used as possible/appropriate.
+
+1. Calculate the calibration constants. For the pixel-by-pixel this
+   would be `DIFC`, (up to) `TZERO` and `DIFA`. The
+   [Gauss-Markov theorem](https://en.wikipedia.org/wiki/Gauss%E2%80%93Markov_theorem)
+   should be used. Deciding which parameters will be used will be
+   determined by using the reduced chi-squared to determine the "best"
+   option. [FindPeakBackground](https://github.com/mantidproject/mantid/blob/v3.5.1/Code/Mantid/Framework/Algorithms/src/FindPeakBackground.cpp#L366)
+   shows a way to compare.
+
+Bank-by-bank
+------------
+
+This calibration determines a multiplicative constant, `need_name`,
+which satisfies the equation (**CHECK THE MATH**)
+
+```
+d_exp = d_obs * need_name
+```
+
+This can then be factored into all of the pixel-by-pixel calibration
+by simple multiplication, e.g. `DIFC_new = DIFC * need_name`. (**OTHER
+EQUATION SHOULD CHANGE TO MAKE THIS CORRECT**)
+
+1. The first three steps are the same as the pixel-by-pixel except for
+   calculations and parameters are in d-spacing.
+
+1. Calculate the calibration constants, `need_name`, which satisfy the
+   equation above. Since it is a scalar, it can be calculated as the
+   average `need_name` for each of the peaks in the observed
+   specturm. This is the 1-parameter version of the Gauss-Markov
+   theorem.
+
+Updating the `idf`
+------------------
+
+Most of this has been done. It involves merging
+[AlignComponents](https://github.com/mantidproject/mantid/compare/master...rosswhitfield:AlignComponent)
```

Comment:
Aha, if I'm following correctly, this new algorithm is a much more general and fancier way of doing what for example for EnginX is done with a simple pixel-by-pixel formula (as they only update the L2 of the individual detectors using the DIFC's). Then you just need the trivial `ApplyCalibration`. 

When this is implemented it might be a good idea to add some notes in these algorithms to clarify
how `ApplyCalibration`, `AlignComponents` and friends interplay, differ and/or overlap.